src/main/java/edu/hm/hafner/analysis/parser/PyLintParser.java

```diff
@@ -13,42 +16,59 @@
  * A parser for the PyLint compiler warnings.
  *
  * @author Sebastian Hansbauer
+ * @author Björn Pedersen
  */
 public class PyLintParser extends FastRegexpLineParser {
     private static final long serialVersionUID = 4464053085862883240L;
 
-    private static final String PYLINT_ERROR_PATTERN = "(.*):(\\d+): \\[(\\D\\d*).*\\] (.*)";
+    // the default pattern matches "--output-format=parseable" output.
+    private static final String DEFAULT_PYLINT_ERROR_PATTERN = "(?<path>.*):(?<line>\\d+): \\[(?<category>\\D\\d*)\\((?<symbol>.*)\\), .*?\\] (?<message>.*)";
 
+
+    public static final String UNKNOWN_CAT = "pylint-unknown";
     /**
      * Creates a new instance of {@link PyLintParser}.
      */
     public PyLintParser() {
-        super(PYLINT_ERROR_PATTERN);
+        super(DEFAULT_PYLINT_ERROR_PATTERN);
+    }
+
+    public PyLintParser(@CheckForNull String regExp) {
```

Comment:
What is the reason for this constructor?

Reply:
To be able to make the regex user-configurable. 

Now used in the legacy test, the changes for warnings-ng plugin will follow later.